enumerables.rb

```diff
@@ -1,41 +1,106 @@
+# frozen_string_literal: true
+# My each method for Ruby, you can use it by simply calling it .my_each
 module Enumerable
-    def my_each
-      return to_enum(:my_each) unless block_given?
-      
-      i = 0
-      array = to_a
-      while i < array.length
-        yield(array[i])
-        i+=1
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    array = to_a
+    while i < array.length
+      yield(array[i])
+      i += 1
     end
     array
   end
- 
 
-def my_each_with_index
+  def my_each_with_index
     return to_enum(:my_each_with_index) unless block_given?
-    
+
     i = 0
     array = to_a
     while i < array.length
       yield(array[i], i)
-      i+=1
+      i += 1
+    end
+    array
   end
-  array
-end
 
-def my_select
+  def my_select
     return to_enum(:my_select) unless block_given?
+
     new_array = []
-  my_each do |x|
-new_array.push(x) if yield(x) == true
+    my_each do |x|
+      new_array.push(x) if yield(x) == true
+    end
+    new_array
   end
-  new_array
-end
-end
-  
 
-puts [1,2,3,4,5,6].my_select(&:even?)
+  def my_all?
+    return to_enum(:my_all?) unless block_given?
 
+    my_each do |x|
+      return false if yield(x) != true
+    end
+    true
+  end
+
+  def my_any?
+    return to_enum(:my_any?) unless block_given?
+
+    my_each do |x|
+      return true if yield(x) == true
+    end
+    false
+  end
+
+  def my_none?
+    return to_enum(:my_none?) unless block_given?
+
+    my_each do |x|
+      return false if yield(x) == true
+    end
+    true
+  end
+
+  def my_count(args = nil)
+    counter = 0
+    if args
+      my_each { |i| counter += 1 if i == args }
+    elsif !block_given?
+      counter = size
+    elsif !args
+      my_each { |i| counter += 1 if yield i }
+    end
+    counter
+  end
+
+  def my_map(&proc_0)
+    return to_enum(:my_map) unless block_given?
+
+    map = []
+    if !proc_0.nil?
+      my_each { |value| map.push(proc_0.call(value)) }
+    elsif block_given?
+      my_each { |value| map.push(yield(value)) }
+    end
+    map
+  end
```

Comment:
### #my_map
- [ ] It executes only the proc when both a block and a proc are given and returns a new array with the results of running the proc once for every element in enum.